docs/source/features/cupti_counters_analysis.rst

```diff
@@ -0,0 +1,80 @@
+For a detailed motivation of the features see `issue 29`_.
+
+
+Collecting CUPTI Counter traces
+-------------------------------
+
+One can collect performance metrics by adding the list of metrics using the experimental config option in PyTorch profiler. Please see `PR94689 <https://github.com/pytorch/pytorch/pull/94689>`_ for more details.
+
+.. code-block:: python
+
+    with torch.profiler.profile(
+        activities=[torch.profiler.ProfilerActivity.CUDA,
+                    torch.profiler.ProfilerActivity.CPU],
+        record_shapes=True,
+        on_trace_ready=trace_handler,
+        experimental_config=torch.profiler._ExperimentalConfig(
+            profiler_metrics=[
+                "kineto__tensor_core_insts",
+                 "dram__bytes_read.sum",
+                 "dram__bytes_write.sum"],
+        profiler_measure_per_kernel=True),
+    ) as prof:
+        res = train_batch(modeldef)
+        prof.step()
+
+Performance measurements are emitted to the trace per kernel.
+  - The events are emitted under a `cuda_profiler_range` category
+  - The counter values are contained inside the *args* json part of the trace.
+
+The CPU operators continue to be emitted as usual. For a full end-end test example try this `gist <https://gist.github.com/briancoutinho/b1ec7919d8ea2bf1f019b4f4cd50ea80>`_.
+
+CUPTI Counter Analyzer
+----------------------
+CUPTI counter trace analyzer can investigate performance measurements per kernel and associate them to operators that the kernel belongs to.
+A single kernel can map to multiple levels of operators (as operators can be nested).
+To represent this we basically provide a list column called `op_stack`.
+For further convenience we add the top and bottom level operator column as well.
+
+The code below runs CUPTI counter analysis on the collected trace.
+
+.. code-block:: python
+
+   analyzer = TraceAnalysis(trace_dir = "/path/to/trace/folder")
+   gpu_kernels = analyzer.get_cupti_counter_data_with_operators(ranks=[0])[0]
+
+This returns a list of dataframes, one per rank or trace file.
+Each dataframe contains the kernel name, op_stack (operator stack), top and bottom level op, and columns for individual performance counters as shown below.
+
+
+.. image:: ../_static/cupti_counter_analysis.png
+
+Example Notebook
+----------------
+For a detailed walkthrough of this feature see the `cupti_flops_analysis
+notebook
+<https://github.com/facebookresearch/HolisticTraceAnalysis/blob/main/examples/cupti_flops_analysis.ipynb>`_
+in the examples folder of the repo.
+
+The trace in this example was collected using `param benchmarks <https://github.com/facebookresearch/param/tree/main/train/compute/python>`_. Run using:
```

Comment:
One of the ways to collect it is using parambenchmarks, open source benchmark tool.